### ISSUE TITLE
<br /> gets duplicated (dev)

### ISSUE THREAD

**Issue by AlekzNet** [Program-O/Program-O#294]
After some recent changes `<br />`s get duplicated.  Here's an example:

```xml
<category>
    <pattern>brtest</pattern>
    <template>
       No br in this line
       One br in this line <br />
       Two br's in this line <br /><br />
    </template>
</category>
```

Result:

<pre>
User: brtest
Bot: No br in this line One br in this line

Two br's in this line

</pre>

HTML:

```html
<span class="username">User: </span>
brtest
<br>
<span class="botname">Bot: </span>
No br in this line One br in this line
<br>
<br>
Two br's in this line
<br>
<br>
<br>
<br>
```

**Comment by AlekzNet** [comment]
From the debug log:

```txt
25-06-2017 20:07:57.20365[4][4] - Elapsed time: 0.007 milliseconds
[parse_aiml_as_XML.php][parseTemplateRecursive][262]
Adding 'No br in this line    One br in this line ' to the response array. tag name is template.
-----------------------
25-06-2017 20:07:57.20366[2][4] - Elapsed time: 0.008 milliseconds
[parse_aiml_as_XML.php][parseTemplateRecursive][201]
Recursively parsing the AIML template.
-----------------------
25-06-2017 20:07:57.20367[4][4] - Elapsed time: 0.009 milliseconds
[parse_aiml_as_XML.php][parseTemplateRecursive][214]
Processing element br at level 1. element XML = <br/>
-----------------------
25-06-2017 20:07:57.20368[4][4] - Elapsed time: 0.008 milliseconds
[parse_aiml_as_XML.php][parseTemplateRecursive][225]
Function parse_html_tag does exist. Processing now.
-----------------------
25-06-2017 20:07:57.20368[4][4] - Elapsed time: 0.005 milliseconds
[parse_aiml_as_XML.php][parseTemplateRecursive][229]
Passing element br to the parse_html_tag function
-----------------------
25-06-2017 20:07:57.20369[2][4] - Elapsed time: 0.006 milliseconds
[parse_aiml_as_XML.php][parse_html_tag][1032]
Parsing a generic HTML tag.
-----------------------
25-06-2017 20:07:57.20370[4][4] - Elapsed time: 0.016 milliseconds
[parse_aiml_as_XML.php][parseTemplateRecursive][234]
Adding '<br/></br/>' to the response array. tag name is br
-----------------------
25-06-2017 20:07:57.20371[2][4] - Elapsed time: 0.007 milliseconds
[parse_aiml_as_XML.php][implode_recursive][137]
Imploding an array into a string. (recursively, if necessary)
-----------------------
25-06-2017 20:07:57.20372[2][4] - Elapsed time: 0.006 milliseconds
[parse_aiml_as_XML.php][implode_recursive][138]
Input: Array
(
    [0] => <br/></br/>
)
```

Somewhere around line 234 in parse_aiml_as_XML.php

**Comment by Dave-Morton** [comment]
@AlekzNet I appreciate your efforts at testing Program O and reporting anything out of the ordinary, but I _DO_ have to mention that you sometimes bring us some weird stuff. 😈 I'll look into it, but probably not until tonight. Still lots of irons in the fire right now.

**Comment by AlekzNet** [comment]
From what I understand, parse_html_tag does not understand single tags like `<br />` and adds  open `<br>` and close '<br>' but without the slash.

**Comment by AlekzNet** [comment]
@patrickschur  So, you just checked and filtered for `br`. What other single tags do we have? `<hr>`, `<img>`?

**Comment by patrickschur** [comment]
@AlekzNet We have a ton of void elements. But in my tests there was only a problem with `<br>`. The other elements works like before... 

If you say that didn't work, I can change that if you want.

**Comment by AlekzNet** [comment]
I mean it can be done in two ways:

1- do not automatically add the closing tag. Just wait for  `</` and add it.
2- create a list of all singe tags and compare the current one with the list

I think the first way is simpler and faster.

**Comment by patrickschur** [comment]
I made a change, I can merge it if you want.

**Comment by AlekzNet** [comment]
Nice one! Works perfectly!

**Comment by Dave-Morton** [comment]
Truthfully, I'd do this a little differently, and set test conditions for other atomic elements as well, such as `<hr/>` and `<img/>`. But that's just me, I guess. ;)

**Comment by patrickschur** [comment]
@Dave-Morton I've already done it? :D

**Comment by Dave-Morton** [comment]
I know, but you're only dealing with the `<br/>` tag. Here's a list of tags that may be similarly affected:

**Comment by patrickschur** [comment]
No... I'm dealing with 'area', 'base', 'br', 'col', 'hr', 'img', 'input', 'link', 'meta', 'param', 'track' and the 'wbr' tag. :P

**Comment by Dave-Morton** [comment]
I was looking at [this](https://github.com/patrickschur/Program-O/commit/eca9bd7497861de5f827da4396a1cdd291b3453e), and didn't see reference to those other tags. My bad.

**Comment by AlekzNet** [comment]
Also, some tags must be added to aiml.xsd to get validated. Will do.

**Comment by Dave-Morton** [comment]
Thanks, @AlekzNet for your help with the validation portion.
